### PR TITLE
Use the v6 version of the actor's bundle.

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- The actor's WASM bundle has moved to v7 to we need to explicitly ask for v6.
- Use `serde_ipld_dagcbor` instead of `cs_serde_cbor` for all encoding/decoding.

**Reference issue to close (if applicable)**

**Other information and links**

### CODE REVIEW

File: blockchain/chain_sync/src/chain_muxer.rs [hunk 1]
```diff
@@ -899,19 +898,20 @@ mod tests {
...
-            encoding::from_slice(&base64::decode("gA==").unwrap()).unwrap();
-
-        assert_eq!(
-            TipsetValidator::compute_msg_root(&blockstore, &usm, &sm)
-                .expect("Computing message root should succeed")
-                .to_string(),
-            "bafy2bzacecmda75ovposbdateg7eyhwij65zklgyijgcjwynlklmqazpwlhba"
-        );
-    }
+    // FIXME: IPLD format changed. Figure out what this test was supposed to do.
```
Comment (LesnyRumcajs) [inline blockchain/chain_sync/src/chain_muxer.rs:901]: Let's have a task for that?

File: blockchain/chain_sync/src/validation.rs [hunk 2]
```diff
@@ -118,15 +118,16 @@ impl<'a> TipsetValidator<'a> {
...
         let secp_cids = secp_msgs
             .iter()
             .map(Cbor::cid)
-            .collect::<Result<Vec<cid_orig::Cid>, fvm_shared::encoding::Error>>()?;
+            .collect::<Result<Vec<Cid>, fvm_shared::encoding::Error>>()?;
 
         // Generate Amt and batch set message values
         let bls_message_root = Amt::new_from_iter(blockstore, bls_cids)?;
         let secp_message_root = Amt::new_from_iter(blockstore, secp_cids)?;
+        dbg!(bls_message_root, secp_message_root);
```
Comment (LesnyRumcajs) [inline blockchain/chain_sync/src/validation.rs]: Is it a leftover from debugging or it is supposed to be here?

File: blockchain/state_manager/src/lib.rs [hunk 3]
```diff
@@ -168,8 +168,7 @@ where
     ) -> anyhow::Result<[u8; 32]> {
         let chain_rand = ChainRand::new(blocks.to_owned(), self.cs.clone(), self.beacon.clone());
         match self.get_network_version(round) {
-            /*NetworkVersion::V15 | */ // FIXME: nv15
-            NetworkVersion::V14 => {
+            NetworkVersion::V15 | NetworkVersion::V14 => {
```
Comment (LesnyRumcajs) [inline blockchain/state_manager/src/lib.rs:171]: :muscle:

File: encoding/src/lib.rs [hunk 4]
```diff
@@ -7,13 +7,14 @@ mod hash;
 
 pub use serde::{de, ser};
 pub use serde_bytes;
-pub use serde_cbor::{error, from_reader, from_slice, tags, to_vec, to_writer};
+//pub use serde_cbor::{error, from_reader, from_slice, tags, to_vec, to_writer};
```
Comment (LesnyRumcajs) [inline encoding/src/lib.rs]: let's remove this?

File: utils/bigint/Cargo.toml [hunk 5]
```diff
@@ -11,13 +11,15 @@ repository = "https://github.com/ChainSafe/forest"
...
+libipld-core = { version = "0.13", features = ["serde-codec"] }
+serde_ipld_dagcbor = "0.1.0"
 num-bigint = "0.4"
 serde = { version = "1.0", features = ["derive"] }
 serde_bytes = { package = "cs_serde_bytes", version = "0.12" }
 num-integer = "0.1"
 
 [dev-dependencies]
-serde_cbor = { package = "cs_serde_cbor", version = "0.12", features = [ "tags" ] }
+#serde_cbor = { package = "cs_serde_cbor", version = "0.12", features = [ "tags" ] }
```
Comment (LesnyRumcajs) [inline utils/bigint/Cargo.toml:22]: let's remove it?

File: vm/actor_interface/src/lib.rs [hunk 6]
```diff
@@ -66,8 +67,8 @@ impl From<StateTreeVersion> for ActorVersion {
             StateTreeVersion::V0 => ActorVersion::V0,
             StateTreeVersion::V1 => ActorVersion::V2,
             StateTreeVersion::V2 => ActorVersion::V3,
-            StateTreeVersion::V3 => ActorVersion::V4,
-            StateTreeVersion::V4 => ActorVersion::V5,
+            StateTreeVersion::V3 => ActorVersion::V6,
+            StateTreeVersion::V4 => ActorVersion::V6,
             //StateTreeVersion::V4 => ActorVersion::V6, // TODO: what version should it point to?
```
Comment (LesnyRumcajs) [inline vm/actor_interface/src/lib.rs:72]: I guess this mystery is already solved?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: blockchain/chain_sync/src/validation.rs (hunk 2 above)
Comment (lemmih) [reply blockchain/chain_sync/src/validation.rs]: A leftover. Should be gone now.

File: utils/bigint/Cargo.toml (hunk 5 above)
Comment (lemmih) [reply utils/bigint/Cargo.toml:22]: Done.

File: encoding/src/lib.rs (hunk 4 above)
Comment (lemmih) [reply encoding/src/lib.rs]: Done.

File: blockchain/chain_sync/src/chain_muxer.rs (hunk 1 above)
Comment (lemmih) [reply blockchain/chain_sync/src/chain_muxer.rs:901]: Done. Tracking link added to the comment.

File: vm/actor_interface/src/lib.rs (hunk 6 above)
Comment (lemmih) [reply vm/actor_interface/src/lib.rs:72]: This is related to state migration / versioning in the native backend. We definitely used the wrong versions in the past but not in a way that mattered. Thankfully all of this code will be deleted soon.

**Review by LesnyRumcajs — APPROVED** [review]
Rock-solid! It's almost 50% fewer files touched than with the original FVM, we're getting there! :smile:

**Review by elmattic — APPROVED** [review]

**Pull request merged**